src/core/Board.ts

```diff
@@ -72,7 +72,20 @@ class Board {
   }
 
   private getPieceMarker(piece: Piece): string {
-    let marker: string = piece.type.toString().charAt(0);
+    const pieceToMarker = new Map([
+      [PieceType.Pawn, 'P'],
+      [PieceType.Knight, 'N'],
+      [PieceType.Queen, 'Q'],
+      [PieceType.King, 'K'],
+      [PieceType.Bishop, 'B'],
+      [PieceType.Rook, 'R']
+    ])
+
+    let marker: (string|undefined) = pieceToMarker.get(piece.type);
```

Comment:
To można zadeklarować poza komponentem, np w cosntants. Dodatkowo wystarczy stworzyć to jako zwykły obiekt readonly a nie Mapę i tym samym zyksac kompletne type safety:

```
const pieceToMarker = {
  [PieceType.Pawn]: 'P',
  [PieceType.Knight]: 'N',
  [PieceType.Queen]: 'Q',
  [PieceType.King]: 'K',
  [PieceType.Bishop]: 'B',
  [PieceType.Rook]: 'R',
} as const;
```  

Wtedy:

` let marker = pieceToMarker[piece.type];` 

ma pewny typ